## Задание (вопрос 3)
Есть сундук, из которого могут выпасть предметы трех типов. Из каждого сундука выпадает ровно один предмет. Вероятности выпадения предметов каждого типа одинаковы. Игрок при успешном прохождении ивента имеет возможность открыть 6 таких сундуков. Игрок собирает полный сет, если у него есть предметы всех трех различных типов. Найти вероятность получения игроком полного сета после успешного прохождения ивента.

In [1]:
import numpy as np

In [2]:
n = 100000
m = 6

Будем считать, что если нам выпадает число в интервале
* (0, 1/3], то это означает выпадение предмета 1-го типа,
* (1/3, 2/3], то это означает выпадение предмета 2-го типа,
* (2/3, 1], то это означает выпадение предмета 3-го типа.

Проведем n экспериментов для m=6 сундуков и посчитаем, сколько раз (обозначим за k) выпадет набор из 3-х различных предметов.

In [3]:
k = 0
for i in np.arange(0, n, 1):
    count = 0
    p = np.random.uniform(0, 1, m)
    if (p[p <= 1/3.0].size > 0):
        count = count + 1
    if (p[(p > 1/3.0) & (p <= 2/3.0)].size > 0):
        count = count + 1
    if (p[p > 2/3.0].size > 0):
        count = count + 1
    if count == 3:
        k = k + 1

Тогда эмпирическая вероятность:

In [4]:
k/n

0.73993

Для проверки сгенерируем всевозможные исходы события и посчитаем их количество.

In [8]:
import itertools

x = [1, 2, 3] 

a = np.array([p for p in itertools.product(x, repeat = 6)])
    
print(len(a))

729


Посчитаем количество благоприятных исходов.

In [9]:
count = 0
for elm in a:
    if set(x).issubset(elm):
        count = count + 1

Тогда теоретическая вероятность:

In [10]:
count/len(a)

0.7407407407407407

## Вывод:
таким образом, вероятность получения игроком полного сета после успешного прохождения ивента составляет 0.74. Данный результат был получен эмпирически и проверен теоретически.